### DISCUSSION TITLE
`getchar` function in rich

### DISCUSSION THREAD

**Discussion by yashrathi-git** [Textualize/rich#1575] [Ideas] [answered]
It would be nice if rich would have `getchar` function just like `click.getchar` (that only gets and returns single char). It would easily allow to create interactive stuff like checkboxes, select menu, and even TUIs using `rich.Live`.

would also solve: #960

**Comment by RonnyPfannschmidt** [comment]
textual is handling input and tu details, rich is solely for formatted printing

**Reply by yashrathi-git** [reply]
I know about textual. But this feature isn't conflicting with what textual does. Rich should have some functionality to get just single char from stdin. This is really easy to implement into rich too.

rich already has methods for `input`, `prompt` etc. so I think this blends well with it.

**Comment by willmcgugan** [comment, marked as answer]
It's a possibility. But my experience of working on Textual, tells me its not as simple as it may seem.

**Reply by yashrathi-git** [reply]
> its not as simple as it may seem

are you talking about implementation of `getchar` function that only inputs single character? Is the implementation textual have not cross-platform? 
I feel like if rich would have some way of responding to keyboard events instantly, like changing something instantly if I press a character it would be really useful. I really love the way `rich.Live`  handles updates.

[click.getchar implementation](https://github.com/pallets/click/blob/65eceb08e392e74dcc761be2090e951274ccbe36/src/click/_termui_impl.py#L624), it's synchronous but cross-platform

**Reply by willmcgugan** [reply]
Nothing to prevent you combining that with Rich…

**Reply by yashrathi-git** [reply]
thanks that's what I have been doing in my projects. ig I would stick to for now. you might consider adding it to rich too.
